flood_adapt/dbs_classes/dbs_template.py

```diff
@@ -0,0 +1,370 @@
+        copy_object.attrs.name = new_name
+        copy_object.attrs.description = new_description
+
+        # Then a save. Checking whether the name is already in use is done in the save function
+        self.save(copy_object)
+
+        # Then save all the accompanied files
+        src = self._path / old_name
+        dest = self._path / new_name
+        for file in src.glob("*"):
+            if "toml" not in file.name:
+                shutil.copy(file, dest / file.name)
+
+    def save(self, object_model: ObjectModel, overwrite: bool = False):
+        """Saves an object in the database. This only saves the toml file. If the object also contains a geojson file,
+        this should be saved separately.
+
+        Parameters
+        ----------
+        object_model : ObjectModel
+            object to be saved in the database
+        overwrite : OverwriteMode, optional
+            whether to overwrite the object if it already exists in the
+            database, by default False
+
+        Raises
+        ------
+        ValueError
+            Raise error if name is already in use.
+        """
+
+        object_exists = object_model.attrs.name in self.list_objects()["name"]
+
+        # If you want to overwrite the object, and the object already exists, first delete it. If it exists and you
+        # don't want to overwrite, raise an error.
+        if overwrite and object_exists:
+            self.delete(object_model.attrs.name, toml_only=True)
+        elif not overwrite and object_exists:
+            raise ValueError(
+                f"'{object_model.attrs.name}' name is already used by another {self._type}. Choose a different name"
+            )
+
+        # If the folder doesnt exist yet, make the folder and save the object
+        if not (self._path / object_model.attrs.name).exists():
+            (self._path / object_model.attrs.name).mkdir()
+
+        object_model.save(
+            self._path / object_model.attrs.name / f"{object_model.attrs.name}.toml"
+        )
+
+    def edit(self, object_model: ObjectModel):
+        """Edits an already existing object in the database.
+
+        Parameters
+        ----------
+        object : ObjectModel
+            object to be edited in the database
+
+        Raises
+        ------
+        ValueError
+            Raise error if name is already in use.
+        """
+        # Check if the object exists
+        if object_model.attrs.name not in self.list_objects()["name"]:
+            raise ValueError(
+                f"'{object_model.attrs.name}' {self._type} does not exist. You cannot edit an object that does not exist."
+            )
+
+        # Check if it is possible to delete the object by saving with overwrite. This then
+        # also covers checking whether the object is a standard object, is already used in
+        # a higher level object or is locked. If any of these are the case, it cannot be deleted.
+        self.save(object_model, overwrite=True)
```

Comment:
If users try to edit objects and get errors that they cannot delete it, I think it will be confusing.
Editing an object in the database is functionally the same as deleting the old and saving the new, but I think adding some duplicated code with more informative errors is okay/good here. 
We could also group the editable/delete-able checks in a function is_mutable() or something and call that twice

Reply:
If you edit, you then still need to delete the files as well. So that would mean that you have to add all the code in the delete function to this new function, or duplicate the code to delete. I now added some text to the error which should make it a bit more descriptive. Also, in the gui, when this error is caught, you can also change the error there based on the function it is called from